### PR TITLE
chore: add binaries for 1.28.13, 1.29.8, 1.30.4

### PR DESCRIPTION
**Pull request by HuShaoRu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add binaries cache for kubernetes patch versions 1.28.13, 1.29.8, 1.30.4

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/components.json [hunk 1]
```diff
@@ -359,14 +359,17 @@
         "v1.28.9-1",
         "v1.28.11",
         "v1.28.12",
+        "v1.28.13",
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/components.json:362]: can we start thinking about removing removing 1.27 patch versions? we need to be cognizant of the space requirements for caching all these versions at once

**Review by cameronmeissner — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/components.json (hunk 1 above)
Comment (haitch) [reply parts/linux/cloud-init/artifacts/components.json:363]: no, we have to keep 1.27 related container images for LTS customers.

Comment (haitch) [reply parts/linux/cloud-init/artifacts/components.json:363]: but I don't think we need this much:
  ~~"v1.27.13",~~
  ~~"v1.27.13-2",~~
  ~~"v1.27.14-1",~~
  "v1.27.15",
   "v1.27.16",

Comment (HuShaoRu) [reply parts/linux/cloud-init/artifacts/components.json:363]: I'll merge this pr first. we might discuss it later and make necessary changes in rp

**Pull request merged**